src/wallets/WalletContractV5.ts

```diff
@@ -0,0 +1,310 @@
+        let state = await provider.getState();
+        if (state.state.type === 'active') {
+            let res = await provider.get('seqno', []);
+            return res.stack.readNumber();
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Get Wallet Extensions
+     */
+    async getExtensions(provider: ContractProvider) {
+        let state = await provider.getState();
+        if (state.state.type === 'active') {
+            const result = await provider.get('get_extensions', []);
+            return result.stack.readCellOpt();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Get Wallet Extensions
+     */
+    async getExtensionsArray(provider: ContractProvider) {
+        const extensions = await this.getExtensions(provider);
+        if (!extensions) {
+            return [];
+        }
+
+        const dict:  Dictionary<bigint, bigint> = Dictionary.loadDirect(
+            Dictionary.Keys.BigUint(256),
+            Dictionary.Values.BigInt(8),
+            extensions
+        );
+
+        return dict.keys().map(key => {
+            const wc = dict.get(key)!;
+            const addressHex = key ^ (wc + 1n);
+            return Address.parseRaw(`${wc}:${addressHex.toString(16)}`);
+        })
+    }
+
+    /**
+     * Get is secret-key authentication enabled
+     */
+    async getIsSecretKeyAuthEnabled(provider: ContractProvider) {
+        let res = await provider.get('is_public_key_enabled', []);
+        const result = res.stack.readNumber();
+        return result !== 0;
+    }
+
+    /**
+     * Send signed transfer
+     */
+    async send(provider: ContractProvider, message: Cell) {
+        await provider.external(message);
+    }
+
+    /**
+     * Sign and send transfer
+     */
+    async sendTransfer(provider: ContractProvider,   args: Wallet5SendArgs & { messages: MessageRelaxed[] }) {
+        const transfer = this.createTransfer(args);
+        await this.send(provider, transfer);
+    }
+
+    /**
+     * Sign and send add extension request
+     */
+    async sendAddExtension(provider: ContractProvider, args: Wallet5SendArgs & { extensionAddress: Address }) {
+        const request = this.createAddExtension(args);
+        await this.send(provider, request);
+    }
+
+    /**
+     * Sign and send remove extension request
+     */
+    async sendRemoveExtension(provider: ContractProvider, args: Wallet5SendArgs & { extensionAddress: Address, }) {
+        const request = this.createRemoveExtension(args);
+        await this.send(provider, request);
+    }
+
+    /**
+     * Sign and send request
+     */
+    async sendRequest(provider: ContractProvider, args: Wallet5SendArgs & { actions: (OutActionSendMsg | OutActionExtended)[], }) {
+        const request = this.createRequest(args);
+        await this.send(provider, request);
+    }
+
+    private createActions( args: {  messages: MessageRelaxed[], sendMode?: Maybe<SendMode> }) {
```

Comment:
maybe specify sendMode for every message? and remove optional? I do not think what using sendMode: PAY_GAS_SEPARATELY + IGNORE_ERRORS as default is a safe practice.